proposals/0002-cxx-attributes.md

```diff
@@ -124,4 +124,65 @@ Below are a few more examples of C++ attributes that we could support:
     }
   }
 ```
+## Detailed design
+
+The simplest explanation of this feature is supporting C++11 attribute syntax on
+all shared grammar elements between HLSL and C++. This spec attempts to detail
+some of the grammar and parsing implications and will specify the process by
+which existing attributes will convert to C++ attributes.
+### Attribute Parsing
+
+This proposal introduces new grammar formulations for parsing attributes
+matching some formulations from C++ `dcl.attr.grammar`. Specifically:
+
+```ebnf
+attribute-specifier-seq:    [ attribute-specifier-seq ], attribute-specifier
+attribute-specifier:        "[" , "[" , attribute-list , "]" , "]"
+attribute-list:             [ attribute ] ||
+                            attribute-list , "," , [attribute] ||
+                            { attribute } ||
+                            attribute-list ,  "," , { attribute }
+attribute:                  attribute-token , [attribute-argument-clause]
+attribute-token:            identifier ||
+                            attribute-scoped-token
+attribute-scoped-token:     attribute-namespace , "::" , identifier
+attribute-namespace:        identifier
+attribute-argument-clause:  "(" , balanced-token-seq , ")"
+balanced-token-seq:         [ balanced-token ] ||
+                            balanced-token-seq , balanced-token
+balanced-token:             "(" , balanced-token-seq , ")"
+                            "[" , balanced-token-seq , "]"
+                            "{" , balanced-token-seq , "}"
+```
+
+In contrast to existing HLSL annotations and Microsoft-style attributes, these
+attribute identifiers are case-sensitive `identifier` tokens.

```

Comment:
This is probably clear enough from context and I'm just being pedantic, but "attribute identifiers are case-sensitive `identifier` tokens" isn't strictly true (there's also the `attribute-scoped-token` formulation).

Maybe:
> In contrast to existing HLSL annotations and Microsoft-style attributes, these formulations use case-sensitive `identifier` tokens